sql/wsrep_high_priority_service.cc

```diff
@@ -126,7 +126,43 @@ static int apply_events(THD*                       thd,
                         const wsrep::const_buffer& data,
                         wsrep::mutable_buffer&     err)
 {
-  int const ret= wsrep_apply_events(thd, rli, data.data(), data.size());
+  static const LEX_CSTRING savepoint= { STRING_WITH_LEN("wsrep_retry") };
+  uint n_retries = 0;
+  bool savepoint_exists = false;
+  uint applier_retry_count = wsrep_applier_retry_count;
+
+  if (applier_retry_count > 0) {
+    /* create a savepoint in case we need to retry applying */
+    savepoint_exists = (FALSE == trans_savepoint(thd, savepoint));
+  }
+
+  int ret= wsrep_apply_events(thd, rli, data.data(), data.size());
+
+  while (ret && savepoint_exists && n_retries < applier_retry_count) {
+    /* applying failed, retry applying events */
+
+    /* rollback to savepoint without telling Wsrep-lib */
+    bool saved_wsrep_on = thd->variables.wsrep_on;
+    thd->variables.wsrep_on = false;
```

Comment:
this thread is replication applier, so it must have wsrep_on == ON, therefore using saved_wsrep_on is redundant

Reply:
Fixed as suggested.